marshal.go

```diff
@@ -180,7 +178,17 @@ func marshalOpenssh(pk interface{}, opts *MarshalOptions) ([]byte, error) {
 		blocksize = aes.BlockSize
 	}
 
-	check := mrand.Uint32()
+	// Get a crypto rand in the range [0, uint32-max)
+	randnum, err := rand.Int(rand.Reader, big.NewInt(0x100000000))
+	if err != nil {
+		return nil, fmt.Errorf("sshkeys: failed to get random number: %w", err)
+	}
+	// Convert it to uint32. As the content was within uint32 limits, nothing should be lost
+	check := uint32(randnum.Uint64())
```

Comment:
I think you can trust the library here.  If you really wanna verify, write a generative test using https://pkg.go.dev/testing/quick that'll run this a bunch of times and you can verify stocahstically you'll never get an out-of-range number.